independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -51,11 +51,12 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         String name = null;
 
         for (AnnotationInstance annotation : beanDeployment.getAnnotations(beanClass)) {
-            if (beanDeployment.getQualifier(annotation.name()) != null) {
+            for (AnnotationInstance qualifierAnnotation : beanDeployment.getQualifierExtractor(annotation.name())
```

Comment:
Hm, maybe we could simplify the code a little bit. `Function<> getQualifierExtractor(annotation.name())` -> `Collection<AnnotationInstance> getQualifiers(annotation)` and the loop could be replaced with `beanDeployment.getQualifiers(annotation).forEach(qualifiers::add);`. WDYT?

Reply:
I just realized that my code doesn't call `continue` properly on the outer loop once we know it's qualifier (so we needn't process it any further). And so does yours. We should take that into account, so in fact it should look something like this:

```
            Collection<AnnotationInstance> qualifierCollection = beanDeployment.getQualifierExtractor(annotation.name()).apply(annotation);
            for (AnnotationInstance qualifierAnnotation : qualifierCollection) {
                // Qualifiers
                qualifiers.add(qualifierAnnotation);
            }
            if (!qualifierCollection.isEmpty()) {
                // we needn't process it further, the annotation was a qualifier (or multiple repeating ones)
                continue;
            }
```